master.cfg

```diff
@@ -1,352 +1,322 @@
 # -*- python -*-
 # ex: set filetype=python:
 
-from buildbot.plugins import (worker, steps, changes, schedulers, util,
-                              reporters)
-
+import os
 import toolz
 import operator
-from ursabot.docker import worker_images
+import warnings
+
+from buildbot.plugins import worker, steps, changes, util, reporters
+
 from ursabot.hooks import GithubHook
+from ursabot.utils import Config, ConfigError, Collection, startswith
+from ursabot.workers import DockerLatentWorker
+from ursabot.reporters import ZulipMailNotifier
+from ursabot.schedulers import AnyBranchScheduler, TryScheduler, ForceScheduler
+from ursabot.builders import (BuildFactory, Builder, UrsabotTest,
+                              ArrowCppTest, ArrowPythonTest,
+                              ArrowCppCondaTest, ArrowPythonCondaTest)
+from ursabot.docker import arrow_images, ursabot_images
 
-try:
-    from local import (PRODUCTION, BUILDBOT_URL, ZULIP, EMAIL, WORKER_PORT,
-                       HOSTS, WEBUI_PORT, AUTH, AUTHZ, DATABASE_URL,
-                       WORKER_PASSWORD, CHANGE_PORT, CHANGE_USER,
-                       CHANGE_PASSWORD, GITHUB_TOKEN, GITHUB_HOOK_SECRET,
-                       TRY_SCHEDULER_PORT, TRY_SCHEDULER_USERPASS)
-except ImportError:
-    from buildbot.www.auth import NoAuth
-    PRODUCTION = False
-    BUILDBOT_URL = 'http://localhost:9011/'
-    DATABASE_URL = 'sqlite:///buildbot.sqlite'
-    EMAIL = 'test@buildbot'
-    ZULIP = None
-    HOSTS = None
-    WORKER_PORT = 9021
-    WEBUI_PORT = 8080
-    AUTH = NoAuth()
-    AUTHZ = util.Authz()
-    GITHUB_TOKEN = None
-    CHANGE_PORT = 9041
-    CHANGE_USER = 'test'
-    CHANGE_PASSWORD = 'testpw'
-    GITHUB_HOOK_SECRET = None
-
-
-print(f'IS PRODUCTION: {PRODUCTION}')
-
-# Hierarchy worker -> builder -> scheduler
+
+# Read ursabot configuration from .toml files, export URSABOT_ENV='test' to use
+# test environment. These configurations are merged upon each other, the
+# default merging order is default.toml <- $URSABOT_ENV.toml <- .secrets.toml
+env = os.environ.get('URSABOT_ENV')
```

Comment:
I'd pick `test` as default and put a warn message about it, e.g. "NOT RUNNING IN PROD".